packages/timeline-state-resolver/src/integrations/shotoku/index.ts

```diff
@@ -178,42 +88,13 @@ export class ShotokuDevice extends DeviceWithState<ShotokuDeviceState, DeviceOpt
 
 		return deviceState
 	}
-	get deviceType() {
-		return DeviceType.SHOTOKU
-	}
-	get deviceName(): string {
-		return 'Shotoku ' + this.deviceId
-	}
-	get queue() {
-		return this._doOnTime.getQueue()
-	}
-	/**
-	 * Add commands to queue, to be executed at the right time
-	 */
-	private _addToQueue(commandsToAchieveState: Array<Command>, time: number) {
-		_.each(commandsToAchieveState, (cmd: Command) => {
-			this._doOnTime.queue(
-				time,
-				undefined,
-				async (cmd: Command) => {
-					return this._commandReceiver(time, cmd.command, cmd.context, cmd.timelineObjId)
-				},
-				cmd
-			)
-		})
-	}
-	/**
-	 * Compares the new timeline-state with the old one, and generates commands to account for the difference
-	 * @param oldState The assumed current state
-	 * @param newState The desired state of the device
-	 */
-	private _diffStates(oldState: ShotokuDeviceState, newState: ShotokuDeviceState): Array<Command> {
+	diffStates(oldState: ShotokuDeviceState | undefined, newState: ShotokuDeviceState): Array<ShotokuCommandWithContext> {
 		// unfortunately we don't know what shots belong to what camera, so we can't do anything smart
 
-		const commands: Array<Command> = []
+		const commands: Array<ShotokuCommandWithContext> = []
 
 		_.each(newState.shots, (newCommandContent: ShotokuDeviceStateContent, index: string) => {
```

Comment:
While you're here, it would be nice to remove the use of underscore. Not essential though